src/Constants/UiConstants.h

```diff
@@ -16,9 +16,9 @@ namespace constants
 
 	const int RESOURCE_BOX_WIDTH = 200;
 
-	const size_t NO_SELECTION = SIZE_MAX;
+	const uint32_t NO_SELECTION = _UI32_MAX;
```

Comment:
Consider using `numeric_limits<uint32_t>::max()` for C++ code. Requires `#include <limits>`.

Reference:
[`std::numeric_limits<T>::max`](https://en.cppreference.com/w/cpp/types/numeric_limits/max)

----

To avoid duplication of the type, this line could become:
```cpp
const auto NO_SELECTION = std::numeric_limits<uint32_t>::max();
```

Could also replace the `const` with `constexpr`, to ensure the value is initialized at compile time, rather than runtime.